compliance/tags/azure_rg_tags/azure_resource_group_tags.pt

```diff
@@ -0,0 +1,314 @@
+      } else {
+        if (_.isEmpty(resource_tags)) {
+          result.push({
+            id: ar["id"],
+            region: ar["region"],
+            name: ar["name"],
+            resource_group: resource_group_name,
+            namespace: namespace,
+            resource_type: resource_type,
+            tag: 'none'
+          })
+        } else {
+          if (_.has(resource_tags, param_tag_key)) {
+            var rg_tag_value = resource_tags[param_tag_key] ;
+            console.log('TAG KEY FOUND')
+            console.log('Tag Value: '+rg_tag_value)
+
+            if (resource_group_name == rg_tag_value) {
+              console.log('Tag Value matches Resource Group Name')
+              result.push({
+                id: ar["id"],
+                region: ar["region"],
+                name: ar["name"],
+                resource_group: resource_group_name,
+                namespace: namespace,
+                resource_type: resource_type,
+                tag: rg_tag_value
+              })
+            } else {
+              console.log('Tag Value differs from Resource Group Name. Adding to Incident..')
+              result.push({
+                id: ar["id"],
+                region: ar["region"],
+                name: ar["name"],
+                resource_group: resource_group_name,
+                namespace: namespace,
+                resource_type: resource_type,
+                tag: rg_tag_value
+              })
+            }
+          } else {
+            console.log('Tag Key Not Found')
+            result.push({
+              id: ar["id"],
+              region: ar["region"],
+              name: ar["name"],
+              resource_group: resource_group_name,
+              namespace: namespace,
+              resource_type: resource_type,
+              tag: 'N/A'
+            })
+          }
+        }
+      }
+    })
+
+EOS
+end
+
+policy "azure_resource_policy" do
+  validate_each $filtered_resources do
+    summary_template "{{ rs_project_name }} (Account ID: {{ rs_project_id }}): {{ len data }} Azure Resources out of compliance"
+    detail_template <<-EOS
+# Azure Resources Incorrectly Tagged
+| Resource Name | Resource Group | Azure Region | Resource ID | Tag Value | 
+| ------------- | -------------- | ------------ | ----------- | --------- |
+{{ range data -}}
+| {{.name}} | {{.resource_group}} | {{.region}} | {{.id}} | {{.tag}}
+{{ end -}}
+EOS
+    escalate $email
+    escalate $tag_resources
+    check eq(val(item, "resource_group"), val(item, "tag"))
+  end
+end
+
+escalation "email" do
+   email $param_email
+end
+
+escalation "tag_resources" do
+  run "tag_resources", data, $param_azure_tenant_id, $param_azure_sub, $param_tag_key
+end
+
+define tag_resources($data, $param_azure_tenant_id, $param_azure_sub, $param_tag_key) return $all_responses do
+
+  call get_access_token($param_azure_tenant_id) retrieve $access_token
+
+  $providers_response = http_request(
```

Comment:
If I remember correctly, this is a very expensive call. Curious if you have any benchmark data on how long this call takes?

Reply:
i do not

Reply:
How about how long the policy takes to run from start to end in an example account?